src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/StartTask.groovy

```diff
@@ -16,17 +16,44 @@
 package net.wasdev.wlp.gradle.plugins.tasks
 
 import org.gradle.api.tasks.TaskAction
+import net.wasdev.wlp.ant.ServerTask;
 
 class StartTask extends AbstractTask {
 
+    protected final String START_APP_MESSAGE_REGEXP = "CWWKZ0001I.*"
+
     @TaskAction
     void start() {
         def params = buildLibertyMap(project);
         params.put('clean', project.liberty.clean)
         if (project.liberty.timeout != null && project.liberty.timeout.length() != 0) {
-        params.put('timeout', project.liberty.timeout)
+            params.put('timeout', project.liberty.timeout)
         }
         executeServerCommand(project, 'start', params)
+        
+        ServerTask serverTask = new ServerTask()
+        serverTask.setInstallDir(params.get('installDir'))
+        serverTask.setServerName(params.get('serverName'))
+        serverTask.setUserDir(params.get('userDir'))
+        serverTask.setOutputDir(params.get('outputDir'))
+        serverTask.initTask()
+        
+        def verifyTimeout = project.liberty.verifyTimeout
+        if(project.liberty.verifyTimeout < 0) {
+            verifyTimeout = 30
+        }
+        long timeout = verifyTimeout * 1000
+        long endTime = System.currentTimeMillis() + timeout;
+        if(project.liberty.applications) {
+            String[] apps = project.liberty.applications.split("[,\\s]+")
+            for(String archiveName : apps) {
+                String verify = serverTask.waitForStringInLog(START_APP_MESSAGE_REGEXP + archiveName, timeout, serverTask.getLogFile())
+                if (!verify) { 
+                    executeServerCommand(project, 'stop', buildLibertyMap(project))
+                    throw new Exception("CWWKM2148E: Unable to verify if the server was started after ${verifyTimeout} seconds")
```

Comment:
Should we use GradleException?  
https://stackoverflow.com/questions/10312259/recommended-way-to-stop-a-gradle-build